app/models/service_template_orchestration.rb

```diff
@@ -24,4 +45,13 @@ def self.default_reconfiguration_entry_point
   def my_zone
     orchestration_manager.try(:my_zone) || MiqServer.my_zone
   end
+
+  def self.validate_config_info(options)
+    config_info = options[:config_info]
+    unless (config_info[:template_id] && config_info[:manager_id]) || (config_info[:template] && config_info[:manager])
```

Comment:
Ideally we should allow the combination. What you have here is likely real. I am ok with it.